list.py

```diff
@@ -0,0 +1,242 @@
+class _Node:
+    def __init__(self, elem, next = None):
+        self.elem = elem
+        self.next = next
+
+
+class _ListIterator:
+    def __init__(self, head):
+        self.current = head
+
+    
+    def __iter__(self):
+        return self
+
+
+    def __next__(self):
+        if self.current == None:
+            raise StopIteration
+
+        elem = self.current.elem
+        self.current = self.current.next
+        return elem
+
+
+class List:
+    def __init__(self):
+        self.__head = None
+        self.__tail = None
+        self.__length = 0
+
+
+    def __len__(self):
+        '''
+        Returns the length of the List.
+        '''
+        return self.__length
+
+
+    def length(self) -> int:
+        '''
+        Returns the length of the List.
+        '''
+        return self.__length
+
+
+    def __eq__(self, other: 'List'):
+        '''
+        Comparing two Lists.
+        '''
+        curr_node = self.__head
+
+        count = 0
+
+        if self.__length != other.length():
+            return False
+
+        while count < self.__length:
+
+            if self[count] != other[count]:
+                return False
+
+            curr_node = curr_node.next
+            count += 1
+
+        return True
+
+    
+    def __str__(self) -> str:
+        '''
+        Creates a string from the List.
+        '''
+        string = '['
+
+        curr_node = self.__head
+
+        while curr_node:
+            string += curr_node.elem + ', ' * (curr_node.next != None) 
+            curr_node = curr_node.next 
+        
+        string += ']'
+
+        return string
+
+    # R
+    def __getitem__(self, index):
+        '''
+        Gets element by index.
+
+        Panics if index >= len.
```

Comment:
Panics in Rust and exceptions in Python are mostly two different things. Let's follow python's terminology everywhere where the word "panic" was mentioned 

```suggestion
        Raises an exception if index >= len.
```

Reply:
Fixed it. Also added small examples: 20b2f82e88bd67efe12dc018d4ae6fa3c04f3a09